### PR TITLE
Adding codecov targets and thresholds

### PR DESCRIPTION
**Pull request by michellethomas**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [x] Build / Development Environment
- [ ] Documentation

### SUMMARY
In [SIP-13](https://github.com/apache/incubator-superset/issues/6132) we discussed having automated checks for code coverage (section [P.5.3]). This PR creates a codecov yaml for us to set the targets and thresholds we want, so if we add a setting in the repo to alert if these rules aren't met, we can change them if we feel they are too strict.

Questions:

Should we have folders we ignore? I initially put in the visualizations folder to ignore because it's been difficult to add unit tests to cover this code, is that what we want? Are there other folders to ignore?

### TEST PLAN
Used codecov's validator `curl --data-binary @.codecov.yml https://codecov.io/validate`
Tested on a PR on in my fork

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@mistercrunch @kristw @john-bodley @xtinec @DiggidyDave

### CODE REVIEW

**Comment by michellethomas** [comment]
After discussing this, it seemed like there was general support for setting a patch target. But in the interest of getting this in sooner rather than later, I set a 1% threshold instead which I believe means it's going to look at the coverage of our repo and flag if the diff decreases by > 1% from that number. And we can increase it over time if we want.

What do you all think about ignoring the visualization folder, do we want to do that?

**Comment by DiggidyDave** [comment]
I was actually going to reach out to you today about this, I'm in favor of it 100%. I _personally_ would actually be pretty draconian about it--at least as a starting point--meaning:
- don't omit any folders
- don't allow any reduction, even <1%

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+coverage:
+  status:
+    project:
+      default:
+        # Commits pushed to master should not make the overall
+        # project coverage decrease by more than 1%:
+        target: auto
+        threshold: 1%
```
Comment (DiggidyDave) [inline .codecov.yml]: as noted in other comment, can we hold a hard line on this and disallow any reduction in coverage?

Also, purely out of curiosity: is there a way to bypass the check (with committer approval) in an emergency situation?

**Review by DiggidyDave — APPROVED** [review]

**Comment by michellethomas** [comment]
The reason it may be useful to ignore the visualizations folder is that it's pretty difficult to add unit tests on that code, but I'm fine to remove the `ignore`.

File: .codecov.yml (hunk 1 above)
Comment (mistercrunch) [reply .codecov.yml]: Nope I think you need admin to override the blockers, which none of the committers get. You'd have to open an apache INFRA ticket :(

Comment (DiggidyDave) [reply .codecov.yml]: It just occurred to me that _this file_ provides a perfectly suitable (and auditable!) mechanism for emergency exemptions... the committer could just add one or more ignore paths with a comment in the file documenting the need, so we would have a written record of anything that accidentally gets left in there

**Comment by michellethomas** [comment]
Set the threshold to 0% based on @DiggidyDave's suggestion and removed ignore directory. Anyone have any final comments? Once we get this in I'll need to get the repo to add this as a commit check. Until then it won't change anything. Seems good to just get it in and then we can adjust as necessary.

**Comment by DiggidyDave** [comment]
SWEEEET

**Review by john-bodley — APPROVED** [review]

**Pull request merged**